### PR TITLE
Stop OmegaConfigLoader from reading config from hidden directory like `ipynb_checkpoints`

### LINKED ISSUE #2583: `OmegaConfigLoader` chokes when `.ipynb_checkpoints` contains checkpoint of `catalog.yml`
## Description
As per title.

## Context
I was migrating a project from `TemplatedConfigLoader` to `OmegaConfigLoader`, and got an error that was not previously happening.

Jupyter automatically creates checkpoints of files (both notebooks and plain text files) in an `.ipynb_checkpoints` directory:

```
$ find conf/base -name "catalog*.yml"
conf/base/.ipynb_checkpoints/catalog-checkpoint.yml
conf/base/catalog.yml
```

## Steps to Reproduce
1. Copy `conf/base/catalog.yml` to `conf/base/.ipynb_checkpoints/catalog-checkpoint.yml`
2. Try `kedro catalog list` while using `TemplatedConfigLoader`, everything works okay.
3. Switch to `OmegaConfigLoader` and try `kedro catalog list`, see error.

## Expected Result
`OmegaConfigLoader` behavior matches the other catalog files when there are duplicates.

```bash
$ kedro catalog list
DataSets in '__default__' pipeline:
  Datasets mentioned in pipeline:
    CSVDataSet:
    - openrepair-0_3-combined
    - openrepair-0_3-events-raw
    - openrepair-0_3-categories
    - openrepair-0_3
DataSets in 'data_processing' pipeline:
  Datasets mentioned in pipeline:
    CSVDataSet:
    - openrepair-0_3-combined
    - openrepair-0_3-events-raw
    - openrepair-0_3-categories
    - openrepair-0_3
```

## Actual Result

```bash
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /Users/juan_cano/.micromamba/envs/kpolars310/bin/kedro:8 in <module>                             │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/kedro/framework/cli/cl │
│ i.py:211 in main                                                                                 │
│                                                                                                  │
│   208 │   """                                                                                    │
│   209 │   _init_plugins()                                                                        │
│   210 │   cli_collection = KedroCLI(project_path=Path.cwd())                                     │
│ ❱ 211 │   cli_collection()                                                                       │
│   212                                                                                            │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/click/core.py:1130 in  │
│ __call__                                                                                         │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/kedro/framework/cli/cl │
│ i.py:139 in main                                                                                 │
│                                                                                                  │
│   136 │   │   )                                                                                  │
│   137 │   │                                                                                      │
│   138 │   │   try:                                                                               │
│ ❱ 139 │   │   │   super().main(                                                                  │
│   140 │   │   │   │   args=args,                                                                 │
│   141 │   │   │   │   prog_name=prog_name,                                                       │
│   142 │   │   │   │   complete_var=complete_var,                                                 │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/click/core.py:1055 in  │
│ main                                                                                             │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/click/core.py:1657 in  │
│ invoke                                                                                           │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/click/core.py:1657 in  │
│ invoke                                                                                           │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/click/core.py:1404 in  │
│ invoke                                                                                           │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/click/core.py:760 in   │
│ invoke                                                                                           │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/click/decorators.py:38 │
│ in new_func                                                                                      │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/kedro/framework/cli/ca │
│ talog.py:56 in list_datasets                                                                     │
│                                                                                                  │
│    53 │                                                                                          │
│    54 │   session = _create_session(metadata.package_name, env=env)                              │
│    55 │   context = session.load_context()                                                       │
│ ❱  56 │   datasets_meta = context.catalog._data_sets  # pylint: disable=protected-access         │
│    57 │   catalog_ds = set(context.catalog.list())                                               │
│    58 │                                                                                          │
│    59 │   target_pipelines = pipeline or pipelines.keys()                                        │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/kedro/framework/contex │
│ t/context.py:236 in catalog                                                                      │
│                                                                                                  │
│   233 │   │   │   KedroContextError: Incorrect ``DataCatalog`` registered for the project.       │
│   234 │   │                                                                                      │
│   235 │   │   """                                                                                │
│ ❱ 236 │   │   return self._get_catalog()                                                         │
│   237 │                                                                                          │
│   238 │   @property                                                                              │
│   239 │   def params(self) -> Dict[str, Any]:                                                    │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/kedro/framework/contex │
│ t/context.py:279 in _get_catalog                                                                 │
│                                                                                                  │
│   276 │   │                                                                                      │
│   277 │   │   """                                                                                │
│   278 │   │   # '**/catalog*' reads modular pipeline configs                                     │
│ ❱ 279 │   │   conf_catalog = self.config_loader["catalog"]                                       │
│   280 │   │   # turn relative paths in conf_catalog into absolute paths                          │
│   281 │   │   # before initializing the catalog                                                  │
│   282 │   │   conf_catalog = _convert_paths_to_absolute_posix(                                   │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/kedro/config/omegaconf │
│ _config.py:168 in __getitem__                                                                    │
│                                                                                                  │
│   165 │   │   │   base_path = str(Path(self.conf_source) / self.base_env)                        │
│   166 │   │   else:                                                                              │
│   167 │   │   │   base_path = str(Path(self._fs.ls("", detail=False)[-1]) / self.base_env)       │
│ ❱ 168 │   │   base_config = self.load_and_merge_dir_config(                                      │
│   169 │   │   │   base_path, patterns, read_environment_variables                                │
│   170 │   │   )                                                                                  │
│   171 │   │   config = base_config                                                               │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/kedro/config/omegaconf │
│ _config.py:272 in load_and_merge_dir_config                                                      │
│                                                                                                  │
│   269 │   │   │   file: set(config.keys()) for file, config in config_per_file.items()           │
│   270 │   │   }                                                                                  │
│   271 │   │   aggregate_config = config_per_file.values()                                        │
│ ❱ 272 │   │   self._check_duplicates(seen_file_to_keys)                                          │
│   273 │   │                                                                                      │
│   274 │   │   if not aggregate_config:                                                           │
│   275 │   │   │   return {}                                                                      │
│                                                                                                  │
│ /Users/juan_cano/.micromamba/envs/kpolars310/lib/python3.10/site-packages/kedro/config/omegaconf │
│ _config.py:311 in _check_duplicates                                                              │
│                                                                                                  │
│   308 │   │                                                                                      │
│   309 │   │   if duplicates:                                                                     │
│   310 │   │   │   dup_str = "\n".join(duplicates)                                                │
│ ❱ 311 │   │   │   raise ValueError(f"{dup_str}")                                                 │
│   312 │                                                                                          │
│   313 │   @staticmethod                                                                          │
│   314 │   def _resolve_environment_variables(config: Dict[str, Any]) -> None:                    │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
ValueError: Duplicate keys found in /Users/juan_cano/Projects/QuantumBlack 
Labs/talk-kedro-polars/conf/base/.ipynb_checkpoints/catalog-checkpoint.yml and /Users/juan_cano/Projects/QuantumBlack 
Labs/talk-kedro-polars/conf/base/catalog.yml: openrepair-0_3, openrepair-0_3-categories, openrepair-0_3-combined, 
openrepair-0_3-events-raw
```

## Workaround

Changing the file patterns is an effective workaround:

```python
# settings.py

CONFIG_LOADER_ARGS = {
    "config_patterns": {
        "catalog": ["catalog.yml", "**/catalog.yml"],
    }
}
```

## Your Environment
Include as many relevant details about the environment in which you experienced the bug:

* Kedro version used (`pip show kedro` or `kedro -V`): 0.18.7
* Python version used (`python -V`): 3.10.9
* Operating system and version: macOS Ventura

### PR DESCRIPTION
**Pull request by noklam**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

Fix #2583 

This is a bug because ConfigLoader and TemplatedConfigLoader does not exhibit the same behavior.

## Development notes

The root cause of this is because `glob.glob` != `fsspec.filesystem.glob`. 
image

I also check
- Path.glob("*") - include hidden by default
- fs.glob("*") - include hidden by default
- glob.glob("*") - ignore hidden by default

See the signature
```python
glob.glob(pathname, *, root_dir=None, dir_fd=None, recursive=False, include_hidden=False)
```
This `include_hidden` argument was only added in Python 3.11 (but it is always the default).

Noted that this is not so trivial because of the inconsistent between different `glob`. fsspec.glob gives you `str` while `Path.glob` gives you `Path`. In addition I need to add `resolve` method because `.` can be confused with `../` which is relative path.

## Implementation
- Add addition step to check if the path is a hidden folder.
Reference of `glob.glob` implemenation: https://github.com/python/cpython/blob/0c79fabb036ccbe71298b141b0a9ff0506c67ae6/Lib/glob.py#L229-L230

# reviewer:

- Please check if I have missed any edge case.
- `../` will be resolve before the check - ideally it should be done in the upstream. I am not sure what's the need for all the `as_posix` call. But I think it's out of the scope so I keep the change minimal. 

It was slightly tricky to deal with `.` because `fsspec.glob` is more limited compare to `glob.glob` or `Path.glob`, but it's necessary to make it work for any filesystem.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: kedro/framework/hooks/manager.py [hunk 1]
```diff
@@ -48,6 +48,8 @@ def _register_hooks(hook_manager: PluginManager, hooks: Iterable[Any]) -> None:
         if not hook_manager.is_registered(hooks_collection):
             hook_manager.register(hooks_collection)
 
+    hook_manager.check_pending() # Validate hook_impl respect hook_spec
```
Comment (ankatiyar) [inline kedro/framework/hooks/manager.py:51]: What's this for?

Comment (noklam) [reply kedro/framework/hooks/manager.py:51]: Ignore it for now, I have no idea maybe I merged from other branches accidentally.

File: RELEASE.md [hunk 2]
```diff
@@ -21,7 +21,8 @@
 * Updated `kedro pipeline create` and `kedro catalog create` to use new `/conf` file structure.
 * Converted `setup.py` in default template to `pyproject.toml` and moved flake8 configuration
   to dedicated file `.flake8`.
-
+* Fixed a bug that `OmegaConfigLoader` read config from hidden directory like `.ipynb_checkpoints`.
```
Comment (ankatiyar) [inline RELEASE.md:24]:
```suggestion
* Updated `OmegaConfigLoader` to ignore config from hidden directories like `.ipynb_checkpoints`.
```

**Review by ankatiyar — APPROVED** [review]
LGTM :)

File: RELEASE.md [hunk 3]
```diff
@@ -21,6 +21,8 @@
 * Updated `kedro pipeline create` and `kedro catalog create` to use new `/conf` file structure.
 * Converted `setup.py` in default template to `pyproject.toml` and moved flake8 configuration
   to dedicated file `.flake8`.
+* Fixed a bug that `OmegaConfigLoader` read config from hidden directory like `.ipynb_checkpoints`.
```
Comment (ankatiyar) [inline RELEASE.md:24]: This wasn't removed btw

Comment (noklam) [reply RELEASE.md]: Thanks, I missed this when I merged from main and get conflicts

**Pull request merged**